plugins/experimental/stale_while_revalidate/stale_while_revalidate.c

```diff
@@ -240,38 +212,52 @@ xstrcmp(const void *a, const void *b)
   return strcmp((const char *)a, (const char *)b);
 }
 
-static void
-parse_response(StateInfo *state)
+static int
+parse_response(TSIOBufferReader reader)
 {
   TSIOBufferBlock block;
-  TSParseResult pr = TS_PARSE_CONT;
-  int64_t avail;
-  char *start;
+  int64_t avail = 0;
+  char *start = NULL;
+  char *srch;
 
-  block = TSIOBufferReaderStart(state->resp_io_buf_reader);
+  block = TSIOBufferReaderStart(reader);
 
-  while ((pr == TS_PARSE_CONT) && (block != NULL)) {
-    start = (char *)TSIOBufferBlockReadStart(block, state->resp_io_buf_reader, &avail);
-    if (avail > 0) {
-      pr = TSHttpHdrParseResp(state->resp_info->parser, state->resp_info->buf, state->resp_info->http_hdr_loc,
-                              (const char **)&start, (const char *)(start + avail));
-    }
-    block = TSIOBufferBlockNext(block);
+  if (block) {
+    start = (char *)TSIOBufferBlockReadStart(block, reader, &avail);
+  }
+
+  if (! start || ! (srch=memchr(start,'\n',avail)) 
```

Comment:
manual parsing of the first-space-on-the-first-line of an HTTP response.   Forward one and we have an atoi() status code.